### PR TITLE
Added support for NGroups operations

### PR DESCRIPTION
**Pull request by saprakas**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by saprakas** [comment]
PR started as an update of the last API version in preview mode. But the checks from step1 suggested to create a new API version.

**Comment by gary-x-li** [comment]
>         "x-ms-long-running-operation": true

If it's a Long Running Operation, your 200 and 201 responses should return proper header instead of `NGroup`. Please make sure you are following the async pattern defined in ARM RPC: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md)

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:955 in 03a730b. [](commit_id = 03a730b1158c71f8b12fa4ab274abd6807ceb1cd, deletion_comment = False)

**Comment by gary-x-li** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/ngroups/{ngroupsName}/stop": {

Should this match the `start` endpoint regarding the response code and `x-ms-long-running-operation`?

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:1089 in 03a730b. [](commit_id = 03a730b1158c71f8b12fa4ab274abd6807ceb1cd, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         }

Both of these have been defined in `Resource`. Why defining them again here?

You should favor TrackedResource from common types: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json)

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:2963 in 03a730b. [](commit_id = 03a730b1158c71f8b12fa4ab274abd6807ceb1cd, deletion_comment = False)

**Comment by gary-x-li** [comment]
>     "UserAssignedIdentityAdditionalProperties": {

Favor common types: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/managedidentity.json)

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:3010 in 03a730b. [](commit_id = 03a730b1158c71f8b12fa4ab274abd6807ceb1cd, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "provisioningState": {

Define provisioningState properly. For example, [https://github.com/Azure/azure-rest-api-specs/blob/2b700e5e84d4a95880d373e6a4bce5d16882e4b5/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ContainerApps.json#L570](https://github.com/Azure/azure-rest-api-specs/blob/2b700e5e84d4a95880d373e6a4bce5d16882e4b5/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ContainerApps.json#L570)

---
Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:3040 in 03a730b. [](commit_id = 03a730b1158c71f8b12fa4ab274abd6807ceb1cd, deletion_comment = False)

**Comment by gary-x-li** [comment]
You should also add examples of the new APIs.

**Comment by shivg7795** [comment]
> You should also add examples of the new APIs.

since these APIs are only for private preview, we are planning to add more detailed examples once these are tested and finalized in the preview

**Comment by shivg7795** [comment]
> > ```
> >     "x-ms-long-running-operation": true
> > ```
> 
> If it's a Long Running Operation, your 200 and 201 responses should return proper header instead of `NGroup`. Please make sure you are following the async pattern defined in ARM RPC: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md
> 
> Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:955 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)

 the given link is not working

**Comment by shivg7795** [comment]
> > ```
> >     }
> > ```
> 
> Both of these have been defined in `Resource`. Why defining them again here?
> 
> You should favor TrackedResource from common types: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json
> 
> Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:2963 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)

The location property from the trackedResources is showing error since its not readonly hence using a resource property here made sense that we can override for patch API

**Comment by shivg7795** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/ngroups/{ngroupsName}/stop": {
> > ```
> 
> Should this match the `start` endpoint regarding the response code and `x-ms-long-running-operation`?
> 
> Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:1089 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)

AFAIK, stop is not considered as a long running operation as containers can be stopped immediately without any long procedure to follow

**Comment by gary-x-li** [comment]
> > > ```
> > >     "x-ms-long-running-operation": true
> > > ```
> > 
> > 
> > If it's a Long Running Operation, your 200 and 201 responses should return proper header instead of `NGroup`. Please make sure you are following the async pattern defined in ARM RPC: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md
> > Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:955 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)
> 
> the given link is not working

Try this one: https://github.com/cloud-and-ai-microsoft/resource-provider-contract

**Comment by gary-x-li** [comment]
> > > ```
> > >     }
> > > ```
> > 
> > 
> > Both of these have been defined in `Resource`. Why defining them again here?
> > You should favor TrackedResource from common types: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json
> > Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:2963 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)
> 
> The location property from the trackedResources is showing error since its not readonly hence using a resource property here made sense that we can override for patch API

You've already defined `location` and `zones` property in `Resource`. Do you intend to override them in `NGroup`?

**Comment by shivg7795** [comment]
> > > > ```
> > > >     }
> > > > ```
> > > 
> > > 
> > > Both of these have been defined in `Resource`. Why defining them again here?
> > > You should favor TrackedResource from common types: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json
> > > Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:2963 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)
> > 
> > 
> > The location property from the trackedResources is showing error since its not readonly hence using a resource property here made sense that we can override for patch API
> 
> You've already defined `location` and `zones` property in `Resource`. Do you intend to override them in `NGroup`?

The location property is marked as mutable in the Resource where for PATCH API we need location to be readonly hence we need to override it

**Comment by shivg7795** [comment]
> > > > ```
> > > >     "x-ms-long-running-operation": true
> > > > ```
> > > 
> > > 
> > > If it's a Long Running Operation, your 200 and 201 responses should return proper header instead of `NGroup`. Please make sure you are following the async pattern defined in ARM RPC: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md
> > > Refers to: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:955 in [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd). [](commit_id = [03a730b](https://github.com/Azure/azure-rest-api-specs/commit/03a730b1158c71f8b12fa4ab274abd6807ceb1cd), deletion_comment = False)
> > 
> > 
> > the given link is not working
> 
> Try this one: https://github.com/cloud-and-ai-microsoft/resource-provider-contract

updated the headers in latest iter

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json [hunk 1]
```diff
@@ -0,0 +1,3195 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NGroup"
+            },
+            "headers": {
```
Comment (gary-x-li) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-09-01-preview/containerInstance.json:1004]: Remove this if the 200 is sync response.

**Review by shivg7795 — APPROVED** [review]

**Pull request merged**